packages/react-relay/modern/ReactRelayRefetchContainer.js

```diff
@@ -184,11 +185,11 @@ function createContainerWithFragments<TBase: ReactClass<*>>(
       this.setState({data: this._resolver.resolve()}, profiler.stop);
     };
 
-    _getFragmentVariables(): Variables {
+    _getFragmentVariables = (): Variables => {
       const {
         getVariablesFromObject,
       } = this.context.relay.environment.unstable_internal;
-      return getVariablesFromObject(
+      return this._localVariables || getVariablesFromObject(
         this.context.relay.variables,
         fragments,
         this.props,
```

Comment:
I noticed that we are not forward variables into descendants , which means, `_localVariables` is only visible to local;
so only root query's variables are visible to its children?